### PR TITLE
CocoaPods Library integration fixed #96

### LINKED ISSUE #96: CocoaPods Library integration
Instead of including source code, switch to linking a prebuilt library and including the library headers.
1) include j2objc-generated-library.a
2) add j2objc headers folder to HEADER_SEARCH_PATH
3) update PROJECT-bridgingheader.h to reference the all the headers within Swift, as defined by an "export interface" sourceSet

@confile - make the changes to output the correct file and then reassign this to...
@advayDev1 - so he can modify the file with the correct output

### PR DESCRIPTION
**Pull request by confile**

### CODE REVIEW

File: src/main/groovy/com/github/brunobowden/j2objcgradle/tasks/J2objcXcodeTask.groovy [hunk 1]
```diff
@@ -96,13 +96,14 @@ class J2objcXcodeTask extends DefaultTask {
...
+                "s.source_files = '${srcDirRelativeToBuildDir}**/*.{h}'\n" +
                 "s.public_header_files = '${srcDirRelativeToBuildDir}**/*.h'\n" +
                 "s.resources = '${j2objcResourceDirName}/**/*'\n" +
                 "s.requires_arc = true\n" +
-                "s.libraries = 'ObjC', 'guava', 'javax_inject', 'jre_emul', 'jsr305', 'z', 'icucore'\n" +
+                "s.preserve_path = '${project.buildDir}/j2objc/*.a'\n" +
+                "s.libraries = 'ObjC', 'guava', 'javax_inject', 'jre_emul', 'jsr305', 'z', 'icucore', 'j2objc-generated-library'\n" +
                 "s.xcconfig = { 'HEADER_SEARCH_PATHS' => '${getJ2ObjCHome()}/include', " +
-                "'LIBRARY_SEARCH_PATHS' => '${getJ2ObjCHome()}/lib' }\n" +
+                "'LIBRARY_SEARCH_PATHS' => '${getJ2ObjCHome()}/lib ${project.buildDir}/j2objc }\n" +
```
Comment (advayDev1) [inline src/main/groovy/com/github/brunobowden/j2objcgradle/tasks/J2objcXcodeTask.groovy]: hmm i don't see how this is working.  there is an unterminated single quote here.

Comment (confile) [inline src/main/groovy/com/github/brunobowden/j2objcgradle/tasks/J2objcXcodeTask.groovy]: thanks fixed

File: src/main/groovy/com/github/brunobowden/j2objcgradle/tasks/J2objcXcodeTask.groovy [hunk 2]
```diff
@@ -96,13 +96,14 @@ class J2objcXcodeTask extends DefaultTask {
...
                 "s.version = '1.0'\n" +
                 "s.summary = 'Generated by the j2objc.gradle plugin.'\n" +
-                "s.source_files = '${srcDirRelativeToBuildDir}**/*.{h,m}'\n" +
+                "s.source_files = '${srcDirRelativeToBuildDir}**/*.{h}'\n" +
                 "s.public_header_files = '${srcDirRelativeToBuildDir}**/*.h'\n" +
                 "s.resources = '${j2objcResourceDirName}/**/*'\n" +
                 "s.requires_arc = true\n" +
-                "s.libraries = 'ObjC', 'guava', 'javax_inject', 'jre_emul', 'jsr305', 'z', 'icucore'\n" +
+                "s.preserve_path = '${project.buildDir}/j2objc/*.a'\n" +
+                "s.libraries = 'ObjC', 'guava', 'javax_inject', 'jre_emul', 'jsr305', 'z', 'icucore', 'j2objc-generated-library'\n" +
```
Comment (brunobowden) [inline src/main/groovy/com/github/brunobowden/j2objcgradle/tasks/J2objcXcodeTask.groovy]: Is this the generated library that doesn't exist yet but that you're expecting?

Comment (confile) [inline src/main/groovy/com/github/brunobowden/j2objcgradle/tasks/J2objcXcodeTask.groovy]: @brunobowden Yes You wrote here: https://github.com/brunobowden/j2objc-gradle/issues/96 that the lib name is ``j2objc-generated-library.a``. I asked several questions there but no answer yet.

**Comment by brunobowden** [comment]
We'll likely change the name but that's fine for now. Please can you squash the commits:
https://github.com/brunobowden/j2objc-gradle/blob/master/CONTRIBUTING.md#preparation

Also, can you look in to how to configure separate debug and release libraries. Is that done with separate library_include_paths for each target? You can do that in another PR.

**Comment by confile** [comment]
@brunobowden squash done 

You want to include something like

```
j2objc-generated-library1.a
j2objc-generated-library2.a
2objc-generated-library3.a
```
if this is what you want you have to specify the libs in ``s.libraries``.

**Comment by brunobowden** [comment]
How do you do separate release and debug library include paths?

**Comment by confile** [comment]
I think this is what you are searching for: https://github.com/CocoaPods/CocoaPods/issues/2847

**Comment by brunobowden** [comment]
@advayDev1 - to confirm, for your system to work, it's the same HEADER_SEARCH_PATHS, but a different LIBRARY_SEARCH_PATHS? 

@confile - that looks like what we want. Can you check if this is the best way to do it. Sounds like we'll need a debug and release podspec's. Lastly, it would be great if the Xcode task could do a "Create folder reference" for the generated code. This is the blue folder that includes the hierarchy and doesn't cause any compilation.

**Comment by advayDev1** [comment]
late to the party here, yes different LIBRARY_SEARCH_PATHS, same HEADER_SEARCH_PATHS

**Pull request merged**